### PR TITLE
feat: add Infinity Wallet

### PR DESCRIPTION
**Pull request by infinitys2025**

### CODE REVIEW

File: packages/rainbowkit/src/wallets/walletConnectors/infinityWallet/infinityWallet.ts
Comment (markdalgleish) [inline packages/rainbowkit/src/wallets/walletConnectors/infinityWallet/infinityWallet.ts:2]: Since this is using wagmi's injected connector, can we avoid adding another dependency?

**Comment by markdalgleish** [comment]
We've just re-worked RainbowKit internals in v0.7 so that wallets can be imported individually, ensuring our bundle size doesn't keep increasing as we continue to add more wallets.

Are you able to give us write access to your fork so that we can get you updated to the new API and apply some changes before merging?

**Comment by markdalgleish** [comment]
Sorry for the delay on this, and thanks for the PR!

We've just re-worked RainbowKit internals in v0.7 so that wallets can be imported individually, ensuring our bundle size doesn't keep increasing as we continue to add more wallets.

Thanks for giving us access to your fork — I've pushed some updates to your PR migrating it to the new API.

File: packages/rainbowkit/src/wallets/walletConnectors/infinityWallet/infinityWallet.ts
Comment (InfinityWallet) [reply packages/rainbowkit/src/wallets/walletConnectors/infinityWallet/infinityWallet.ts:2]: Hi @markdalgleish, as Infinity Wallet is not a browser extension but the first standalone web3 browser wallet, the function `openInfinityWallet` is used both to detect if Infinity Wallet is installed on click and to open the DApp in the Infinity Wallet from any browser.

If you would like, rather than importing the dependency for this we could directly integrate the code into the repository?

Is there anything else required to merge this?

**Comment by InfinityWallet** [comment]
> Sorry for the delay on this, and thanks for the PR!
> 
> We've just re-worked RainbowKit internals in v0.7 so that wallets can be imported individually, ensuring our bundle size doesn't keep increasing as we continue to add more wallets.
> 
> Thanks for giving us access to your fork — I've pushed some updates to your PR migrating it to the new API.

Hi @markdalgleish, we we wondering if there was anything else needed to merge this PR? Thanks

File: packages/rainbowkit/package.json
Comment (DanielSinclair) [inline packages/rainbowkit/package.json:63]: Our current approach here is to rely solely on `@wagmi/connectors` for peer dependency connectors. Please first work with the `wagmi` team for upstream support, and then we'll revisit. https://github.com/wagmi-dev/references/tree/main/packages/connectors

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

File: packages/rainbowkit/src/wallets/walletConnectors/infinityWallet/infinityWallet.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/infinityWallet/infinityWallet.ts:53]: Please also provide clarity on this `getUri`, and the desktop equivalent. These should not return undefined for WalletConnect-supporting wallets.

**Comment by DanielSinclair** [comment]
Closing this for now. Can revisit if there is a better integration path.

**Pull request closed without merging**
